code/modules/projectiles/pins.dm

```diff
@@ -0,0 +1,208 @@
+/*
+Alright boys, Firing pins. hopefully with minimal shitcode.
+"pin_auth(mob/living/user)" is the check to see if it fires, put the snowflake code here. return one to fire, zero to flop. ezpz
+
+Firing pins as a rule can't be removed without replacing them, blame a really shitty mechanism for it by NT or something idk, this is to stop people from just taking pins from like a capgun or something.
+*/
+
+/obj/item/device/firing_pin
+	name = "electronic firing pin"
+	desc = "A small authentication device, to be inserted into a firearm receiver to allow operation. NT safety regulations require all new designs to incorporate one."
+	icon = 'icons/obj/firingpins.dmi'
+	icon_state = "firing_pin"
+	item_state = "pen"
+	origin_tech = list(TECH_MATERIAL = 2, TECH_COMBAT = 2)
+	flags = CONDUCT
+	w_class = 1
+	attack_verb = list("poked")
+	var/emagged = FALSE
+	var/fail_message = "<span class='warning'>INVALID USER.</span>"
+	var/selfdestruct = 0 // Explode when user check is failed.
+	var/force_replace = 0 // Can forcefully replace other pins.
+	var/pin_replaceable = 0 // Can be replaced by any pin.
+	var/durable = FALSE //is destroyed when it's pried out with a screwdriver, see gun.dm
+	var/obj/item/weapon/gun/gun
+
+
+
+/obj/item/device/firing_pin/Initialize(mapload)
+	.=..()
+	if(istype(loc, /obj/item/weapon/gun))
+		gun = loc
+
+/obj/item/device/firing_pin/afterattack(atom/target, mob/user, proximity_flag)
+	if(proximity_flag)
+		if(istype(target, /obj/item/weapon/gun))
+			var/obj/item/weapon/gun/G = target
+			if(G.pin && (force_replace || G.pin.pin_replaceable))
+				G.pin.forceMove(get_turf(G))
+				G.pin.gun_remove(user)
+				to_chat(user, "<span class ='notice'>You remove [G]'s old pin.</span>")
+
+			if(!G.pin)
```

Comment:
This should probably be an `else if`? Because:
* Last if removes pin, this sets `G.pin = null`
* This if activates if `G.pin == null`.